codalab/server/bundle_manager.py

```diff
@@ -445,11 +439,19 @@ def get_sort_key(worker):
             else:
                 deps = set(worker['dependencies'])
                 num_available_deps = len(needed_deps & deps)
-            worker_id = worker['worker_id']
 
-            # if the bundle doesn't request GPUs (only request CPUs), prioritize workers that don't have GPUs
-            gpu_priority = bundle_resources.gpus or not has_gpu[worker_id]
```

Comment:
I guess the previous implementation just checked if GPU's have been requested. With your new fix, we want bundles to use CPU workers as much possible, unless GPU's are requested. Am I understanding this correctly?

Reply:
Yes, you are right! Basically there are two cases as follows:
1. bundle.request_gpus > 0
2. bundle.request_gpus == 0

In the first case, when a bundle requests for GPUs, since the logic already filters out CPU workers in [Line400-Line425](https://github.com/codalab/codalab-worksheets/blob/master/codalab/server/bundle_manager.py#L400-L425) before reaching to the sorting algorithm, adding logic to filter GPU workers out takes no effect in the final result. 

The situation that `gpu_priority` would have a real impact is in the second case when a bundle requests for CPUs only. In such a situation, the bundle can be scheduled to run on both GPU workers and CPU workers. However, since GPUs are considered as more precious computing resources than CPUs, we would want to utilize CPU workers as much as possible in the first place.